website/files/models/osfstorage.py

```diff
@@ -108,14 +108,15 @@ def is_checked_out(self):
         return self.checkout is not None
 
     def delete(self, user=None, parent=None):
-        if self.node.preprint_file == self:
-            self.node._is_preprint_orphan = True
-            self.node.save()
+        if self.node.preprint_file == self and not self.node._has_abandoned_preprint:
```

Comment:
And let's go ahead and add the field to the API v2 File Serializer when this is updated.